server/handler/query.go

```diff
@@ -138,6 +138,58 @@ func Query(db *sql.DB) RequestProcessFunc {
 			return nil, err
 		}
 
-		return serializer.NewQueryResponse(tableData, columnNames), nil
+		return serializer.NewQueryResponse(tableData, columnNames, columnTypes), nil
+	}
+}
+
+// columnsInfo returns the column names and column types, or error
+func columnsInfo(rows *sql.Rows) ([]string, []string, error) {
+	names, err := rows.Columns()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	types, err := rows.ColumnTypes()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	typesStr := make([]string, len(types))
+	for i, colType := range types {
+		typesStr[i] = colType.DatabaseTypeName()
+	}
+
+	return names, typesStr, nil
+}
+
+// unmarshallUAST tries to cast data as [][]byte and unmarshall uast nodes
+func unmarshallUAST(data interface{}) ([]*uast.Node, error) {
+	var protobufs [][]byte
+	if err := json.Unmarshal(*data.(*[]byte), &protobufs); err != nil {
+		return nil, err
+	}
+
+	nodes := make([]*uast.Node, len(protobufs))
+
+	for i, v := range protobufs {
+		node := uast.NewNode()
+		if err := node.Unmarshal(v); err != nil {
+			return nil, err
+		}
+		nodes[i] = node
+	}
+
+	return nodes, nil
+}
+
+// addLimit adds LIMIT to the query, performing basic tests to skip it
+// for DESCRIBE TABLE, SHOW TABLES, and avoid '; limit'
+func addLimit(query string, limit int) string {
+	upper := strings.ToUpper(query)
+	if strings.Contains(upper, "DESCRIBE") || strings.Contains(upper, "SHOW") {
+		return query
```

Comment:
please consider to rewrite it like this:

```go
func addLimit(query string, limit int) string {
  query = strings.TrimRight(strings.TrimSpace(query), ";")
  if strings.HasPrefix(strings.ToUpper(query), "SELECT") {
    return fmt.Sprintf("%s LIMIT %d", query, limit)
  }
  return query
```

it covers more cases and imo easier to read

Reply:
Makes sense, changed in 28fc331